File: Arrays/Two_Number_Sum/Java/twoNumSumTwoPointers.java
import java.util.*;

public class TwoNumSumTwoPointers {
  public static void main(String[] args) {
    int[] numbers = new int[] {1, 2, 4, 5, -5, -2};
    int targetSum = 2;

    twoNumSum(numbers, targetSum);
  };

  public static int[] twoNumSum(int[] numbers, int targetSum) {
    Arrays.sort(numbers);
    int left = 0;
    int right = numbers.length - 1;
    
    while (left < right) {
      int currentSum = numbers[left] + numbers[right];

      if (currentSum == targetSum) {
        int[] result = new int[] {numbers[left], numbers[right]};
        System.out.println(Arrays.toString(result));
        System.out.println("You got the answer!");
        return result;
      } else if (currentSum < targetSum) {
        left ++;
      } else if (currentSum > targetSum) {
        right --;
      }
    }
    return new int[0];
  }
}

File: Arrays/Two_Number_Sum/TypeScript/twoNumSumTwoPointers.ts
export function twoNumSumTwoPointers(numbers: number[], targetSum: number) {
  numbers.sort((a,b) => a-b);
  let left: number = 0;
  let right: number = numbers.length - 1

  while (left < right) {
    let num1: number = numbers[left];
    let num2: number = numbers[right];
    let currentSum: number = num1 + num2;
    if (currentSum == targetSum) {
      let result = [num1, num2];
      console.log(result);
      console.log("You got it!");
      return result;
    } else if (currentSum < targetSum) {
      left ++;
    } else if (currentSum > targetSum) {
      right --;
    }
  }
  return [];
}

let numbers: number[] = [1, 2, 4, 5, -5, -2];
let targetSum: number = 2;

twoNumSumTwoPointers(numbers, targetSum);
